### ISSUE TITLE
[BUG] Strange error on Medium, but not on Strong preset

### ISSUE THREAD

**Issue by brandonsturgeon** [levno-710/Prometheus#31]
**Describe the bug**
I was getting an error when using the Medium preset, but not the Strong preset. 

The error:
```
Tried to Add Child that is referencing a Higher non global variable!
```

**Expected behavior**
Given the same input, all presets should display the same behavior (all succeeding, or all failing).

**To Reproduce**
I simplified the code down to the simplest form that still displayed this bug:
```lua
do end
function a() end
math.min(0, 1)
```

If I remove any one of these lines, the Medium preset works fine.

**Screenshots**
![image](https://user-images.githubusercontent.com/7936439/156221818-34b6abc7-dfbb-4aba-8608-83f5c5cb6611.png)

I'll try to investigate later, but wanted to put up an issue in case you already knew the root cause.

**Additional context**
I'm using the `master` branch with the fix from this issue applied: https://github.com/levno-710/Prometheus/issues/26

**Comment by levno-710** [comment]
I haven't had this exact issue, but i will try to find out where it originates from.
For your investigation: The error is thrown in line 204 of ```root/src/prometheus/scope.lua```

**Comment by levno-710** [comment]
This issue had two causes:
The first one was a bug in visitast.lua, that i fixed with 949b4eafb7a6131318fe83438b4757cd23c55dd1.
The second one was a bug in scope.lua that i fixed with 06e4f740fa38c16a0007ce0d519ecd6c93a3bf7d
For me this script does work on medium after applying these changes
